Model/Kount/EnsConfig.php

```diff
@@ -0,0 +1,362 @@
+use Magento\Sales\Model\Order\Invoice;
+use Magento\Sales\Model\Order\Payment;
+use Magento\Sales\Model\ResourceModel\Order\Invoice\Collection;
+use Magento\Sales\Model\Service\CreditmemoService;
+use Magento\Store\Model\ScopeInterface;
+use Magento\Store\Model\StoreManagerInterface;
+
+/**
+ * Class EnsConfig - Contains methods for dealing with Kount ENS Notifications.
+ */
+class EnsConfig
+{
+    const CONFIG_KOUNT_ID = 'payment/braintree/kount_id';
+    const CONFIG_ALLOWED_IPS = 'payment/braintree/kount_allowed_ips';
+    const CONFIG_ENVIRONMENT = 'payment/braintree/kount_environment';
+
+    const RESPONSE_DECLINE = 'D';
+    const RESPONSE_APPROVE = 'A';
+    const RESPONSE_REVIEW = 'R';
+    const RESPONSE_ESCALATE = 'E';
+
+    /**
+     * @var StoreManagerInterface
+     */
+    private $storeManager;
+    /**
+     * @var ScopeConfigInterface
+     */
+    private $scopeConfig;
+    /**
+     * @var OrderInterface
+     */
+    private $order;
+    /**
+     * @var BraintreeAdapter
+     */
+    private $braintreeAdapter;
+    /**
+     * @var TransactionFactory
+     */
+    private $transactionFactory;
+    /**
+     * @var CreditmemoFactory
+     */
+    private $creditmemoFactory;
+    /**
+     * @var CreditmemoService
+     */
+    private $creditmemoService;
+    /**
+     * @var OrderRepositoryInterface
+     */
+    private $orderRepository;
+
+    /**
+     * @param StoreManagerInterface $storeManager
+     * @param ScopeConfigInterface $scopeConfig
+     * @param OrderInterface $order
+     * @param OrderRepositoryInterface $orderRepository
+     * @param BraintreeAdapter $braintreeAdapter
+     * @param TransactionFactory $transactionFactory
+     * @param CreditmemoFactory $creditmemoFactory
+     * @param CreditmemoService $creditmemoService
+     */
+    public function __construct(
+        StoreManagerInterface $storeManager,
+        ScopeConfigInterface $scopeConfig,
+        OrderInterface $order,
+        OrderRepositoryInterface $orderRepository,
+        BraintreeAdapter $braintreeAdapter,
+        TransactionFactory $transactionFactory,
+        CreditmemoFactory $creditmemoFactory,
+        CreditmemoService $creditmemoService
+    ) {
+        $this->scopeConfig = $scopeConfig;
+        $this->storeManager = $storeManager;
+        $this->order = $order;
+        $this->braintreeAdapter = $braintreeAdapter;
+        $this->transactionFactory = $transactionFactory;
+        $this->creditmemoFactory = $creditmemoFactory;
+        $this->creditmemoService = $creditmemoService;
+        $this->orderRepository = $orderRepository;
+    }
+
+    /**
+     * @return array
+     */
+    public function getAllowedIps(): array
+    {
+        $ips = $this->scopeConfig->getValue(self::CONFIG_ALLOWED_IPS);
+        return $ips ? explode(',', $ips) : [];
+    }
+
+    /**
+     * @param $ip
+     * @param $range
+     * @return bool
+     */
+    private function isIpInRange($ip, $range): bool
```

Comment:
I have a rough idea whats going on here and I assume the maths is sound but I'm thinking it might be a tricky one to debug potentially. Would it be possible to add a comment/summary of the process for people debugging in future?

Reply:
I'll try add some comments. This came from an outside source.